File: packages/dataframe/rust/stats/distributions/exponential.test.R
#!/usr/bin/env Rscript

# Exponential distribution test script
# Usage: Rscript exponential.test.R <function> <x_or_p> <rate> [lower_tail] [log_p]

source("test-helpers.R")

# Define exponential distribution functions
exponential_density <- function(x, rate, log = FALSE) {
  dexp(x, rate, log = log)
}

exponential_cumulative <- function(x, rate, lower.tail = TRUE, log.p = FALSE) {
  pexp(x, rate, lower.tail = lower.tail, log.p = log.p)
}

exponential_quantile <- function(p, rate, lower.tail = TRUE, log.p = FALSE) {
  qexp(p, rate, lower.tail = lower.tail, log.p = log.p)
}

exponential_random <- function(n, rate) {
  rexp(n, rate)
}

# Run the test
run_distribution_test(exponential_density, exponential_cumulative, exponential_quantile, exponential_random)

File: packages/dataframe/rust/stats/distributions/exponential.test.ts
#!/usr/bin/env -S deno run --allow-read --allow-run

import { callR, printResults, runComparison } from "./test-helpers.ts";

async function main() {
  console.log("Testing R vs Rust exponential functions...\n");

  const testCases = [
    { func: "dexp", distribution: "exponential", args: [1, 1, 0] },
    { func: "pexp", distribution: "exponential", args: [1, 1, 1, 0] },
    { func: "qexp", distribution: "exponential", args: [0.5, 1, 1, 0] },
  ];

  const results = await runComparison(testCases);
  printResults(results);

  // Test rexp separately since it's not available in Rust interface
  console.log("\nR-only tests:");
  try {
    const rexpResult = await callR("rexp", "exponential", 1, 1);
    console.log(`rexp(1) = ${rexpResult}`);
  } catch (error) {
    console.log(`rexp: Error - ${(error as Error).message}`);
  }
}

if (import.meta.main) {
  main().catch(console.error);
}
